### PR TITLE
0061 reward impl

### LINKED ISSUE #61: Rewards implementation

### PR DESCRIPTION
**Pull request by ciurana-numo**
Resolves #61

### CODE REVIEW

File: coronado/healthmonitor.py [hunk 1]
```diff
@@ -45,7 +47,9 @@ def __init__(self):
     def check(klass) -> object:
         response = super().list()
 
-        return TripleObject(json.loads(response.content))
+        healthCheck = TripleObject(json.loads(response.content))
+        healthCheck.coronadoVersion = __VERSION__
```
Comment (bruceadams) [inline coronado/healthmonitor.py:51]: I don't understand why this is here. `__VERSION__` is globally available. Why return it as an attribute here?

File: coronado/reward.py [hunk 2]
```diff
@@ -86,6 +105,131 @@ def list(klass: object, paramMap = None, **args) -> list:
...
+
+        endpoint = '/'.join([ klass._serviceURL, 'partner/rewards.%s' % action ])
+        response = requests.request('POST', endpoint, headers = klass.headers, json = spec)
+
+        if response.status_code == 200:
+            result = _assembleDetailsFrom(response.content)
+        elif response.status_code == 404:
+            result = False
+        elif response.status_code == 422:
+            # TODO: THIS 404! {"detail":"No reward found for transaction_id \"129\" and offer_id \"bogus-offer-id\"."}
```
Comment (bruceadams) [inline coronado/reward.py:135]: I'm not so sure a 404 is appropriate if either the transaction or the offer are found. Not finding the combination feels different than just _not found_.

**Review by bruceadams — APPROVED** [review]

File: coronado/healthmonitor.py (hunk 1 above)
Comment (ciurana-numo) [reply coronado/healthmonitor.py:51]: For people who don't know Python well or are too lazy to look it up.  Think garden variety bank developer.  They asked for it during the call on Friday.

File: coronado/reward.py (hunk 2 above)
Comment (ciurana-numo) [reply coronado/reward.py:135]: Matt and I have been talking about this for a while (we have a call planned for Monday to wrap).  The current semantics here are 404 given the error message; both IDs make a unique compound key, 404 (not found) would apply for that specific combination.  We'll figure it out on Monday, this is a note to address that during our chat.  Let me know if you want to /join :)

**Pull request merged**